sdk/communication/azure-communication-callautomation/azure/communication/callautomation/_call_automation_client.py

```diff
@@ -0,0 +1,416 @@
+        callback_url: str,
+        **kwargs
+    ) -> CallResult:
+        """
+        Create a call connection request from a source identity to a target identity.
+
+        :param call_invite: Required. Call invitee's information.
+        :type call_invite: CallInvite
+        :param callback_url: Required. The call back url for receiving events.
+        :type callback_url: str
+        :param source_caller_id_number: The source caller Id, a phone number,
+         that's shown to the PSTN participant being invited.
+         Required only when calling a PSTN callee.
+        :type source_caller_id_number: PhoneNumberIdentifier
+        :param source_display_name: Display name of the call if dialing out to a pstn number.
+        :type source_display_name: str
+        :param source_identity: The identifier of the source of the call.
+        :type source_identity: CommunicationIdentifier
+        :param operation_context: A customer set value used to track the answering of a call.
+        :type operation_context: str
+        :param media_streaming_configuration: Media Streaming Configuration.
+        :type media_streaming_configuration: MediaStreamingConfiguration
+        :param azure_cognitive_services_endpoint_url: The identifier of the Cognitive Service resource
+     assigned to this call.
+        :type azure_cognitive_services_endpoint_url: str
+
+        """
+
+        if not call_invite:
+            raise ValueError('call_invite cannot be None.')
+        if not callback_url:
+            raise ValueError('callback_url cannot be None.')
+
+        media_streaming_config = kwargs.pop(
+            "media_streaming_configuration", None)
+        create_call_request = CreateCallRequest(
+            targets=[serialize_identifier(call_invite.target)],
+            callback_uri=callback_url,
+            source_caller_id_number=serialize_phone_identifier(
+                call_invite.sourceCallIdNumber) if call_invite.sourceCallIdNumber else None,
+            source_display_name=call_invite.sourceDisplayName,
+            source_identity=serialize_identifier(
+                self.source_identity) if self.source_identity else None,
+            operation_context=kwargs.pop("operation_context", None),
+            media_streaming_configuration=media_streaming_config.to_generated(
+            ) if media_streaming_config else None,
+            azure_cognitive_services_endpoint_url=kwargs.pop(
+                "azure_cognitive_services_endpoint_url", None),
+        )
+        repeatability_request_id = kwargs.pop("repeatability_request_id", None)
+        repeatability_first_sent = kwargs.pop("repeatability_first_sent", None)
+
+        result = self._client.create_call(
+            create_call_request=create_call_request,
+            repeatability_first_sent=repeatability_first_sent,
+            repeatability_request_id=repeatability_request_id,
+            **kwargs)
+
+        return CallResult(
```

Comment:
Is it possible to return CreateCallResult here? (extending callresult)
The reason is, EventProcessor will utilize different call result for different usage. If answer/createcall shares same return object, it will be hard to implement EventProcessor in the future.